### PR TITLE
Don't try to connect to oneself

### PR DESCRIPTION
**Pull request by gumb0**
Fixes https://github.com/ethereum/aleth/issues/5199

### CODE REVIEW

File: libp2p/Host.cpp [hunk 1]
```diff
@@ -592,6 +592,9 @@ void Host::connect(std::shared_ptr<Peer> const& _p)
     if (!m_run)
         return;
     
+    if (_p->id == id())
+        return;
```
Comment (chfast) [inline libp2p/Host.cpp]: Add comment, and maybe a log?

**Comment by gumb0** [comment]
@chfast I moved the check up the stack, so that we don't even add our own nodeID to NodeTable; and added a log.

File: libp2p/Host.cpp [hunk 2]
```diff
@@ -527,6 +527,12 @@ void Host::addNode(NodeID const& _node, NodeIPEndpoint const& _endpoint)
         else
             return;
 
+    if (_node == id())
+    {
+        cnetdetails << "Ingoring the request to connect to host " << _node;
```
Comment (axic) [inline libp2p/Host.cpp]: host -> self ?

Comment (gumb0) [reply libp2p/Host.cpp]: changed

**Pull request merged**